src/textual/_config.py

```diff
@@ -0,0 +1,207 @@
+from __future__ import annotations
+
+from pathlib import PurePath
+from typing import Iterable, Collection
+
+import tomli
+
+
+class Config:
+    def __init__(
+        self,
+        namespace: str,
+        default_config_path: str | PurePath,
+        user_config_paths: Iterable[str | PurePath] = None,
+    ):
+        """Manages access to configuration for the Textual application.
+        Textual applications may ship with a `textual.toml` file containing default configuration values.
+        We refer to these as "packaged defaults", and they may only contain relating to the Textual app.
+        Users may have a `textual.toml` file on their system which may override these default configuration values.
+        Instantiating this class does not result in disk access. Disk access only occurs when the `resolve` method is
+        called.
+
+        Args:
+            namespace (str): The namespace of this application. Used to filter out irrelevant config.
+            default_config_path (str | PurePath): The path to the default config for this app. This type of config is
+                handled separately from user configuration as it's more restrictive. For example, you cannot bundle
+                a config file with your application that alters global configuration.
+            user_config_paths (Iterable[str | PurePath]): The paths to the config files to load on the users machine.
+                If items appear multiple times in the supplied config files and have matching namespaces, the last
+                occurrence will take priority.
+        """
+        self.namespace = namespace
+        self._default_config_path = default_config_path
```

Comment:
That was my original approach but I scrapped it. The packaged defaults need to be treated quite differently from user configuration at various points so it helps to keep them separate. For example, if you include a default `textual.toml` file with your app, it needs to be filtered to only contain the config relevant to your app. Otherwise you could alter the config for other textual apps, or even global config.